### PR TITLE
Update keys.c

### PR DESCRIPTION
**Pull request by galaxiesFarApart**
2022-03-01: insert "memset" where ELLIP_CURVES are evaluated to re-initialize the "oid_input_HEX" array to all zero; left as it was, multiple different ECC curve evaluations may leave remnants of earlier read input values which can cause "memcpy" to fail (for "invalid/undefined/unmatched values") -> segfault; most notable when evaluating "keys" with a plethora of packets with different ECC algorithms

### CODE REVIEW

**Comment by galaxiesFarApart** [comment]
Ah  - phooey. I didn't really mean to close PR#34 (having problems determining status of created PRs)

**Comment by galaxiesFarApart** [comment]
Attached are dump outputs and test key file that results in segfault (which really occurs in "printf" statement). Same files have also been sent through e-mail.

[ggp234-ecc-Brainpool-P256-test-2022-03-01.pubkey.zip](https://github.com/kazu-yamamoto/pgpdump/files/8213760/ggp234-ecc-Brainpool-P256-test-2022-03-01.pubkey.zip)
[Multiple_ECC_algos_with_many_packets-segfault-2022-03-01.log.zip](https://github.com/kazu-yamamoto/pgpdump/files/8213761/Multiple_ECC_algos_with_many_packets-segfault-2022-03-01.log.zip)
[Multiple_ECC_algos_with_many_packets-with_memset-2022-03-01.log.zip](https://github.com/kazu-yamamoto/pgpdump/files/8213762/Multiple_ECC_algos_with_many_packets-with_memset-2022-03-01.log.zip)

File: keys.c [hunk 1]
```diff
@@ -125,6 +125,7 @@ new_Public_Key_Packet(int len)
 		break;
 	case 18:/*ECDH*/
 		oidLEN = Getc();
+		memset(oid_input_HEX,0,10);
```
Comment (kazu-yamamoto) [inline keys.c:128]: I would like to avoid hard-coding a magic number. Could you use `oid_input_HEX_size` instead of `10`?

**Review by kazu-yamamoto — CHANGES_REQUESTED** [review]
Sorry for the delay. And nice catch!

**Review by kazu-yamamoto — APPROVED** [review]
Now LTGM

**Comment by kazu-yamamoto** [comment]
Cherry-picked and merged. Thank you for your contribution!

**Comment by galaxiesFarApart** [comment]
Yes - using "size" can be done. I had thought of not using 10 realizing that that value could change in the future, but I already had that value established from RFCs. I will work on the change.

**Pull request closed without merging**